experimental/packages/opentelemetry-sdk-metrics-base/src/Instruments.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import * as metrics from '@opentelemetry/api-metrics';
+import { Meter } from './Meter';
+
+export enum InstrumentType {
+    COUNTER = 'COUNTER',
+    HISTOGRAM = 'HISTOGRAM',
+    UP_DOWN_COUNTER = 'UP_DOWN_COUNTER',
+    OBSERVABLE_COUNTER = 'OBSERVABLE_COUNTER',
+    OBSERVABLE_GAUGE = 'OBSERVABLE_GAUGE',
+    OBSERVABLE_UP_DOWN_COUNTER = 'OBSERVABLE_UP_DOWN_COUNTER',
+}
+
+export class Instrument {
+    constructor(private _meter: Meter, private _name: string, private _version?: string) { }
+
+    getName(): string {
+        return this._name;
+    }
+
+    /**
+     * This only exists to stop typescript complaining about unused variable and may be removed. 
+     */
+    getVersion(): string | undefined {
+        return this._version;
+    }
+
+    aggregate(value: number, attributes: metrics.Attributes = {}, ctx: api.Context = api.context.active()) {
+        this._meter.aggregate(this, {
+            value,
+            attributes,
+            context: ctx,
+        });
+    }
+}
+
+export class UpDownCounter extends Instrument implements metrics.Counter {
```

Comment:
```suggestion
export class UpDownCounter extends Instrument implements metrics.UpDownCounter {
```

Reply:
Should Counter and UpDownCounter be separate interfaces? I thought they were only distinguished by configuration.

Reply:
As far as I can tell, they have the same API, the only difference is that UpDownCounter allows negative values, while the "normal" counter just allows positives (this is enforced in the counter below). So it seems fine to me to use the same underlying type.